src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs

```diff
@@ -161,6 +161,24 @@ private void AddBodyParamsAndHeaders(IDictionary<string, string> additionalBodyP
             {
                 _oAuth2Client.AddHeader(PKeyAuthConstants.DeviceAuthHeaderName, PKeyAuthConstants.DeviceAuthHeaderValue);
             }
+
+            AddExtraHttpHeaders();
+        }
+
+        private void AddExtraHttpHeaders()
+        {
+            if (_requestParams.ExtraHttpHeaders != null)
+            {
+                foreach (KeyValuePair<string, string> pair in _requestParams.ExtraHttpHeaders)
+                {
+                    _oAuth2Client.AddHeader(pair.Key, pair.Value);
```

Comment:
Can `Value` be null? Do we need a check? Maybe a check in the `AddHeader` method?